velox/exec/Window.cpp

```diff
@@ -294,18 +417,52 @@ void Window::callResetPartition(vector_size_t partitionNumber) {
   }
 }
 
-std::pair<vector_size_t, vector_size_t> Window::findFrameEndPoints(
-    vector_size_t /*i*/,
-    vector_size_t partitionStartRow,
-    vector_size_t /*partitionEndRow*/,
-    vector_size_t /*peerStartRow*/,
-    vector_size_t peerEndRow,
-    vector_size_t /*currentRow*/) {
-  // TODO : We handle only the default window frame in this code. Add support
-  // for all window frames subsequently.
-
-  // Default window frame is Range UNBOUNDED PRECEDING CURRENT ROW.
-  return std::make_pair(partitionStartRow, peerEndRow);
+void Window::computeVariableFrameBounds(
+    std::vector<vector_size_t>& constantPrecedingFrameOffset,
+    std::vector<vector_size_t>& constantFollowingFrameOffset) {
+  auto updateFrameVector =
+      [&](vector_size_t idx,
+          FrameArg frameArg,
+          std::vector<vector_size_t>& constantFrameOffset) -> void {
+    if (frameArg.constantValue) {
+      auto constantOffset =
```

Comment:
This value can be determined at initialization time itself and made a member variable to use here directly.

Reply:
Thanks for the suggestion, added it as a parameter to windowFrame struct.